File: frontend/src/app/controller/service/admin/chequeffet/InstrumentAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {InstrumentDto} from 'src/app/controller/model/chequeffet/Instrument.model';
import {InstrumentCriteria} from 'src/app/controller/criteria/chequeffet/InstrumentCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class InstrumentAdminService extends AbstractService<InstrumentDto, InstrumentCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/instrument/');
    }

    public constrcutDto(): InstrumentDto {
        return new InstrumentDto();
    }

    public constrcutCriteria(): InstrumentCriteria {
        return new InstrumentCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/easystock/domain/service/facade/admin/chequeffet/InstrumentAdminService.java
package ma.zs.easystock.domain.service.facade.admin.chequeffet;

import java.util.List;
import ma.zs.easystock.domain.model.chequeffet.Instrument;
import ma.zs.easystock.infra.dao.criteria.core.chequeffet.InstrumentCriteria;
import ma.zs.easystock.zynerator.service.IService;





public interface InstrumentAdminService extends  IService<Instrument,InstrumentCriteria>  {

    List<Instrument> findByTypeInstrumentId(Long id);
    int deleteByTypeInstrumentId(Long id);
    long countByTypeInstrumentId(Long id);
    List<Instrument> findByChequeEffetId(Long id);
    int deleteByChequeEffetId(Long id);
    long countByChequeEffetNumeroChequeEffet(String numeroChequeEffet);



}
